cube/src/main/java/zone/cogni/asquare/cube/convertor/data2shacl/ShaclGenerator.java

```diff
@@ -116,16 +122,31 @@ private void addPrefixes(@Nonnull Map<String, String> prefixes,
   }
 
   private void addTypes(Configuration configuration, RdfStoreService rdfStoreService, Model shacl) {
-    List<String> types = getTypes(rdfStoreService);
+    List<String> types = getTypes(configuration, rdfStoreService);
```

Comment:
so, I guess types should be sorted by priority based on order in `prioritisedNamespaces`.

also note: `getTypes` not only sort but also log if there is ambiguity. like eli:Language vs jolux:Language. by default the ambiguity should be logged as a warning.

but I would even consider a configuration option which is more strict where you would throw an exception.